### PR TITLE
WIP: Prometheus

### PR DESCRIPTION
**Pull request by aniketfuryrocks**

### CODE REVIEW

File: prometheus.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+global:
```
Comment (riordanp) [inline prometheus.yml:1]: This file is not required for the deployment, prometheus is already configured

File: src/workers/prometheus.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        format!(
+            "HTTP/1.1 200 OK\r\nContent-Length: {}\r\n\r\n{}",
+            payload.len(),
+            payload
+        )
+    }
+
+    async fn handle_stream(&self, stream: &mut TcpStream) -> anyhow::Result<()> {
+        let metrics = self.metrics_capture.get_metrics().await;
+        let metrics = serde_prometheus::to_string(&metrics, None, HashMap::new())?;
```
Comment (riordanp) [inline src/workers/prometheus.rs:31]: Please add "literpc" as a prefix so we can namespace metrics https://github.com/w4/serde_prometheus/blob/master/src/lib.rs#L89

File: src/workers/prometheus.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        stream.flush().await?;
+
+        Ok(())
+    }
+
+    pub fn sync(self) -> JoinHandle<anyhow::Result<()>> {
+        #[allow(unreachable_code)]
+        tokio::spawn(async move {
+            let listener = TcpListener::bind("[::]:9500").await?;
```
Comment (riordanp) [inline src/workers/prometheus.rs:46]: You can uncomment the metrics block in fly.toml and change the ports to match which will enable prometheus polling

**Review by riordanp — CHANGES_REQUESTED** [review]
Unfortunately serde_prometheus doesn't seem to support metric types. We'll have to see how these appear in grafana and might need a different solution if it's not working.

File: prometheus.yml (hunk 1 above)
Comment (aniketfuryrocks) [reply prometheus.yml:1]: Removed

File: src/workers/prometheus.rs (hunk 2 above)
Comment (aniketfuryrocks) [reply src/workers/prometheus.rs:31]: Done

File: src/workers/prometheus.rs (hunk 3 above)
Comment (aniketfuryrocks) [reply src/workers/prometheus.rs:46]: Fixed

**Pull request closed without merging**